beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -940,19 +947,31 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         let connected_peer_count = self.network_globals.connected_peers();
         if connected_peer_count > self.target_peers {
             //remove excess peers with the worst scores, but keep subnet peers
-            for (peer_id, _) in self
+            //must also ensure that the outbound-only peer count does not go below the minimum threshold
+            let mut peers_removed = 0;
+            let mut n_outbound_removed = 0;
+            for (peer_id, info) in self
                 .network_globals
                 .peers
                 .read()
                 .worst_connected_peers()
                 .iter()
                 .filter(|(_, info)| !info.has_future_duty())
-                .take(connected_peer_count - self.target_peers)
-                //we only need to disconnect peers with healthy scores, since the others got already
-                //disconnected in update_peer_scores
-                .filter(|(_, info)| info.score_state() == ScoreState::Healthy)
             {
-                disconnecting_peers.push(**peer_id);
+                if peers_removed == connected_peer_count - self.target_peers {
+                    break;
+                }
+                if info.is_outbound_only() {
+                    if n_outbound_removed < outbound_only_peer_count - min_outbound_only_target {
+                        n_outbound_removed += 1;
```

Comment:
you count this guy as removed, but there is one extra check a couple of lines after that would alter this count

Reply:
I meant to mimic the behavior of the original code, which kept ```connected_peer_count - self.target_peers``` peers and then further filtered the list based on whether they were healthy or not. Since ```update_peer_scores``` would have already disconnected the unhleathy peers, it should increment the count, even if we don't add it to the disconnecting_peers list.